tasks/setup.py

```diff
@@ -228,3 +229,57 @@ def download_go_tools(ctx) -> SetupResult:
         status = Status.FAIL
 
     return SetupResult("Download Go tools", status)
+
+
+vscode_config_template = """{{
+    "go.buildTags": "{build_tags}",
+    "go.lintTool": "golangci-lint",
+    "go.lintFlags": [
+        "--build-tags",
+        "{build_tags}",
+        "--config",
+        "{workspace_folder}/.golangci.yml"
+    ],
+    "[go]": {{
+        "editor.formatOnSave": true,
+    }},
+    "gopls": {{
+        "formatting.local": "github.com/DataDog/datadog-agent",
+        "directoryFilters": {excluded_directories},
+    }},
+    "go.testTimeout": "0s",
+    "go.testFlags": [
+      "-v"
+    ],
+    "ruff.configurationPreference": "filesystemFirst",
+    "shellcheck.customArgs": [
+        "--severity=info", "--shell=bash"
+    ],
+    "shellcheck.exclude": [
+        "SC2059",
+        "SC2028"
+    ],
+    "shellcheck.run": "onSave"
+}}"""
+
+
+@task
+def vscode_settings(_):
+    print(color_message("Creating initial VSCode setting file...", Color.BLUE))
+    if os.path.exists(".vscode/settings.json"):
+        print(color_message("VSCode settings file already exists. Skipping...", Color.ORANGE))
+        user_input = input("Do you want to overwrite it? (y/N)")
+        if user_input.lower() != "y":
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** You can use this: https://github.com/DataDog/datadog-agent/blob/2963b0f24cc9cca2ffc12695e891a59ba4fea3cf/tasks/libs/common/user_interactions.py#L4-L19